File: app/src/main/java/io/oldering/tvfoot/red/di/module/MainActivityModule.java
package io.oldering.tvfoot.red.di.module;

import android.app.Activity;
import dagger.Binds;
import dagger.Module;
import dagger.android.ActivityKey;
import dagger.android.AndroidInjector;
import dagger.multibindings.IntoMap;
import io.oldering.tvfoot.red.di.component.MainActivitySubComponent;
import io.oldering.tvfoot.red.util.MainActivity;

@Module(subcomponents = MainActivitySubComponent.class) //
public abstract class MainActivityModule {
  @Binds @IntoMap @ActivityKey(MainActivity.class) //
  abstract AndroidInjector.Factory<? extends Activity> bindMainActivityInjectorFactory(
      MainActivitySubComponent.Builder builder);
}

File: app/src/main/java/com/benoitquenaudon/tvfoot/red/injection/module/MainActivityModule.kt
package com.benoitquenaudon.tvfoot.red.injection.module

import android.app.Activity
import com.benoitquenaudon.tvfoot.red.app.domain.main.MainActivity
import dagger.Binds
import dagger.Module

@Module
abstract class MainActivityModule {
  @Binds abstract fun provideActivity(activity: MainActivity): Activity
}